plugins/modules/ec2_instance.py

```diff
@@ -1463,20 +1463,23 @@ def change_network_attachments(instance, params):
 
 def find_instances(ids=None, filters=None):
     paginator = client.get_paginator('describe_instances')
+    sanitized_filters = dict()
+
     if ids:
         params = dict(InstanceIds=ids)
     elif filters is None:
         module.fail_json(msg="No filters provided when they were required")
     else:
         for key in list(filters.keys()):
             if not key.startswith("tag:"):
-                filters[key.replace("_", "-")] = filters.pop(key)
-        params = dict(Filters=ansible_dict_to_boto3_filter_list(filters))
+                sanitized_filters[key.replace("_", "-")] = filters[key]
```

Comment:
Needs an "else" to ensure that tag based filters are also added to the list of sanitized filers